timelock-impl/src/test/java/com/palantir/atlasdb/timelock/AsyncTimelockServiceImplTest.java

```diff
@@ -19,21 +19,180 @@
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
 
+import com.codahale.metrics.MetricRegistry;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.palantir.atlasdb.futures.AtlasFutures;
+import com.palantir.atlasdb.timelock.api.GenericNamedMinTimestamp;
+import com.palantir.atlasdb.timelock.api.NamedMinTimestampLeaseResponse;
 import com.palantir.atlasdb.timelock.lock.AsyncLockService;
 import com.palantir.atlasdb.timelock.lock.LockLog;
+import com.palantir.atlasdb.timelock.lockwatches.BufferMetrics;
 import com.palantir.atlasdb.timelock.lockwatches.RequestMetrics;
+import com.palantir.timestamp.InMemoryTimestampService;
 import com.palantir.timestamp.ManagedTimestampService;
+import com.palantir.tritium.metrics.registry.DefaultTaggedMetricRegistry;
+import java.util.UUID;
+import java.util.concurrent.ScheduledExecutorService;
+import org.jmock.lib.concurrent.DeterministicScheduler;
+import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+public final class AsyncTimelockServiceImplTest {
```

Comment:
Some bits in this tests are a bit over-specified, the timestamp service can return anything as long as it respects monotonicity. The one we use here just returns increments of 1 and I use that to assert things return as expected

Reply:
Hmm yeah I think this is quite fragile, we should probably do the range math / checks on inequalities 😬